File: FootyFocus-Backend/src/main/java/com/example/FootyFocus/entity/LeagueStandings.java
package com.example.FootyFocus.entity;

import jakarta.persistence.*;
import jakarta.persistence.Entity;
import jakarta.transaction.Transactional;
import lombok.Getter;
import lombok.Setter;

@Entity
@Transactional
@Setter
@Getter

@Table(name = "leagueStandings")
public class LeagueStandings {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    //for competition - Primier league, Ligue 1, la liga
    //under competition
    private String competitionName;
    private String competitionEmblem;

    //for season start and end date
    private String seasonStartDate;
    private String seasonEndDate;
    private int currentMatchDay;

    //For actual standings
    //from table
    private int position;

    //from team
    private String teamName;
    private String teamCrest;
    private String teamShortName;

    //from table
    private int matchesPlayed;
    private  String teamForm;
    private int gamesWon;
    private int gamesLost;
    private int gamesDrew;
    private int noOfPoints;
    private int goalDifference;



}

File: FootyFocus-Frontend/src/app/models/leagueStandings.ts
// export interface LeagueStandings{  // an interface for getting the nested information within the json file
    
//     id: number ;
//     competition:{
    
//         competitionName: string;
//         competitionEmblem: string;
//     },

//     season:{
//         seasonStartDate: string;
//         seasonEndDate: string;
//         currentMatchDay: string;
//     },

//     standings:{
        
//         table: {
//             position: number;
            
//             team:{
//                 teamName: string;
//                 teamShortName: string;
//                 teamCrest: string;
//             },
//             matchesPlayed: number;
//             teamForm: string;
//             gamesWon: number;
//             gamesLost: number;
//             gamesDrew: number;
//             noOfPoints: number;
//             goalDifference: number;
//         }
//     }
//}

export interface LeagueStandings{

    id: number | null;
  competitionName: string;
  competitionEmblem: string;
  seasonStartDate: string;
  seasonEndDate: string;
  currentMatchDay: number;
  position: number;
  teamName: string;
  teamCrest: string;
  teamShortName: string;
  matchesPlayed: number;
  teamForm: string;
  gamesWon: number;
  gamesLost: number;
  gamesDrew: number;
  noOfPoints: number;
  goalDifference: number;
}
